vortex-flatbuffers/flatbuffers/vortex-serde/message.fbs

```diff
@@ -9,8 +9,28 @@ enum Compression: uint8 {
     None = 0,
 }
 
+/// An ArrayData describes the hierarchy of an array as well as the locations of the data buffers that appear
+/// immediately after the message in the byte stream.
+table ArrayMessage {
+    /// The array's hierarchical definition.
+    array: Array;
+    /// The row count of the array.
+    row_count: uint64;
+    /// The locations of the data buffers of the array, in ascending order of offset.
+    buffers: [Buffer];
+}
+
+table Buffer {
```

Comment:
maybe this should have a better name, since it's really the metadata describing a buffer rather than the pile of bytes itself